test/python/circuit/test_circuit_operations.py

```diff
@@ -393,6 +396,26 @@ def test_repeat(self):
             rep = qc.repeat(3)
             self.assertEqual(rep, ref)
 
+    @data('gate', 'instruction')
+    def test_repeat_appended_type(self, subtype):
+        """Test repeat appends Gate if circuit contains only gates and Instructions otherwise."""
+        sub = QuantumCircuit(2)
+        sub.x(0)
+
+        if subtype == 'gate':
+            sub = sub.to_gate()
+        else:
+            sub = sub.to_instruction()
+
+        qc = QuantumCircuit(2)
+        qc.append(sub, [0, 1])
+        rep = qc.repeat(3)
+
+        if subtype == 'gate':
+            self.assertTrue(all(isinstance(op[0], Gate) for op in rep.data))
+        else:
+            self.assertTrue(all(isinstance(op[0], Instruction) for op in rep.data))
+
```

Comment:
Following #4520 , this test will always generate a `Gate` (but the `'instruction'` case will still pass since `Gate` subclasses `Instruction`).